src/interfaces/vault.ts

```diff
@@ -330,11 +330,16 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
       if (token === EthAddress) {
         throw new SdkError("deposit:v2:eth not implemented");
       } else {
+        const shouldUsePartner = this.yearn.services.partner && this.yearn.services.partner.isAllowed(vault);
         const vaultContract = new Contract(vault, VaultAbi, signer);
+
         const makeTransaction = async (overrides: CallOverrides) => {
-          const tx = await vaultContract.populateTransaction.deposit(amount, overrides);
+          const tx = shouldUsePartner
+            ? await this.yearn.services.partner!.populateDepositTransaction(vault, amount, overrides)
+            : await (vaultContract as Contract).populateTransaction.deposit(amount, overrides);
```

Comment:
I think the `vaultContract` is already of type `Contract` so no need to cast it

```suggestion
            : await vaultContract.populateTransaction.deposit(amount, overrides);
```

Reply:
oh you are right, this was a leftover for other fix I had to do

Reply:
@FiboApe it would be nice to have the commit that addresses this in here, so I know if I can resolve this conversation or not. Usually 1 commit per comment/suggestion and then added here so the reviewer knows that it was addressed would be great. What do you think? It's hard for me to see how was this addressed

Reply:
solved on 64bdcb84424cdca052f18fbae628b30e089b0076